packages/gatsby-plugin-image/src/components/__tests__/gatsby-image.browser.tsx

```diff
@@ -0,0 +1,209 @@
+import React from "react"
+import { GatsbyImage, ISharpGatsbyImageData } from "../gatsby-image.browser"
+import { render, waitFor } from "@testing-library/react"
+import * as hooks from "../hooks"
+
+type GlobalOverride = NodeJS.Global &
+  typeof global.globalThis & {
+    GATSBY___IMAGE: boolean
+    SERVER: boolean
+  }
+
+// Prevents terser for bailing because we're not in a babel plugin
+jest.mock(`../../../macros/terser.macro`, () => (strs): string => strs.join(``))
+
+describe(`GatsbyImage browser`, () => {
+  let beforeHydrationContent: HTMLDivElement
+  let image: ISharpGatsbyImageData
+
+  beforeEach(() => {
+    console.warn = jest.fn()
+    ;(global as GlobalOverride).SERVER = true
+    ;(global as GlobalOverride).GATSBY___IMAGE = true
+  })
+
+  beforeEach(() => {
+    image = {
+      width: 100,
+      height: 100,
+      layout: `fluid`,
+      images: { fallback: { src: `some-src-fallback.jpg` } },
+      placeholder: { sources: [] },
+      sizes: `192x192`,
+      backgroundColor: `red`,
+    }
+
+    beforeHydrationContent = document.createElement(`div`)
+    beforeHydrationContent.innerHTML = `
```

Comment:
Woah, is this to replicate the SSR? Clever.

Reply:
Nice